src/org/jitsi/impl/neomedia/rtp/RTPEncodingImpl.java

```diff
@@ -0,0 +1,469 @@
+     */
+    int getIndex()
+    {
+        return idx;
+    }
+
+    /**
+     * Returns a boolean that indicates whether or not this
+     * {@link RTPEncodingImpl} depends on the subjective quality index that is
+     * passed as an argument.
+     *
+     * @param idx the index of this instance in the track encodings array.
+     * @return true if this {@link RTPEncodingImpl} depends on the subjective
+     * quality index that is passed as an argument, false otherwise.
+     */
+    boolean requires(int idx)
+    {
+        if (idx < 0)
+        {
+            return false;
+        }
+
+        if (idx == this.idx)
+        {
+            return true;
+        }
+
+
+        boolean requires = false;
+
+        if (!ArrayUtils.isNullOrEmpty(dependencyEncodings))
+        {
+            for (RTPEncodingImpl enc : dependencyEncodings)
+            {
+                if (enc.requires(idx))
+                {
+                    requires = true;
+                    break;
+                }
+            }
+        }
+
+        return requires;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean matches(byte[] buf, int off, int len)
+    {
+        long ssrc = RawPacket.getSSRCAsLong(buf, off, len);
+
+        if (primarySSRC != ssrc && rtxSSRC != ssrc)
+        {
+            return false;
+        }
+
+        if (temporalId == -1)
+        {
+            return true;
+        }
+
+        int tid = track.getMediaStreamTrackReceiver()
+            .getStream().getTemporalID(buf, off, len);
+
+        return tid == temporalId;
+    }
+
+    /**
+     * Gets a boolean flag that indicates whether or not this instance is
+     * streaming or if it's suspended.
+     *
+     * @param active true if this {@link RTPEncodingImpl} is active, otherwise
+     * false.
+     */
+    void setActive(boolean active)
+    {
+        this.active = active;
+    }
+
+    /**
+     *
+     * @param pkt
+     * @param nowMs
+     *
+     * @return the {@link FrameDesc} that was updated, otherwise null.
+     */
+    FrameDesc update(RawPacket pkt, long nowMs)
+    {
+        // Update rate stats.
+        rateStatistics.update(pkt.getLength(), nowMs);
+
+        long ts = pkt.getTimestamp();
+        FrameDesc frame = frames.get(ts);
+
+        if (frame == null)
+        {
+            synchronized (frames)
+            {
+                frames.put(ts, frame = new FrameDesc(this, ts));
+            }
+
+            // We measure the stable bitrate on every new frame.
+            lastStableBitrateBps = getBitrateBps(nowMs);
+        }
+
+        // Update the frame description.
+        boolean frameChanged = frame.update(pkt);
+        if (frameChanged)
+        {
+            // Frame boundaries heuristics.
+
+            // Find the closest next frame.
+            Map.Entry<Long, FrameDesc> ceilingEntry
+                = frames.ceilingEntry((ts + 1) & 0xFFFFFFFFL);
```

Comment:
If we are close to a wrap, but not at 2^32-1 yet, then ts + 1 will still be big, but the next frame might be already wrapped, and we would fail to find it. Is the only result of this just a failure to infer the frame boundaries (i.e. to optimize)?